solar_consumer/data/nighttime.py

```diff
@@ -0,0 +1,134 @@
+# solar_consumer/data/nighttime.py
+from __future__ import annotations
+
+from typing import Union, Optional
+import pandas as pd
+import pvlib
+import os
+from loguru import logger
+
+
+# Load GSP lat/lon for night-time zeroing from CSV (no datamodel dependency)
+DIR = os.path.dirname(__file__)
+
+
+def _load_gsp_locations() -> pd.DataFrame | None:
+    """Load GSP lat/lon if the CSV exists; return None otherwise."""
+    candidates = [
+        os.path.join(DIR, "uk_gsp_locations_20250109.csv"),
+        os.path.join(DIR, "data", "uk_gsp_locations_20250109.csv"),
+    ]
+    for path in candidates:
+        if os.path.exists(path):
+            df = pd.read_csv(path)
+            if "gsp_id" in df.columns:
+                return df.set_index("gsp_id")
+    logger.warning("GSP locations CSV not found; skipping night-time zeroing.")
+    return None
+
+
+_GSP_LOCATIONS = _load_gsp_locations()
+
+# NOTE:
+# rather than computing irradiance or using cloud info,
+# treat any hour where the sun elevation < ~5° as night.
+# pvlib's solarposition is well validated; this is equivalent
+# to OCF's metnet threshold but avoids needing extra columns.
+
+
+def make_night_time_zeros(
+    df: pd.DataFrame,
+    *,
+    latitude: Optional[float] = None,
+    longitude: Optional[float] = None,
+    gsp_id: Optional[int] = None,
+    ts_col: str = "target_time_utc",
+    mw_col: str = "generation_mw",
```

Comment:
could this be renamed, `generation_col` and `timestamp_col`, that'll make it slightly easier to read